chrome-extension/popup.js

```diff
@@ -127,37 +129,125 @@ function downloadButtonCallback(chapter_list_url) {
 var download_count = 0;
 var expected_download_count = 0;
 
+
+function getBookNameArray(urls)
+{
+  var names = "";
+
+  for (var i = 0 ; i < urls.length ; i++)
+    names += i+1 + ". " + getBookNameFromUrl(urls[i]) + "\n";
+
+  return names;
+}
+
+function isInQuenePage(url)
+{
+  var re = new RegExp('(\.+://www.safaribooksonline.com/s/)\.*');
+  var match = re.exec(url);
+
+  var re2 = new RegExp('(\.+://www.safaribooksonline.com/s/\\?limit=100)\.*');
+  var match2 = re2.exec(url);
+
+  if(match != null && match2 != null)
+    return 2;     // return 2 if the page has loaded 100 books in the Safari Queue page
+  else if (match != null)
+    return 1;     // return 1 if the page is in Safari Queue page
+  else 
+    return 0;     // return 0 if the page is NOT in Safari Queue page
+}
+
+function clickedEvent(url)
+{
+  var re = new RegExp('(\.+://www.safaribooksonline.com/library/view/\.+/\\d+/)\.*');
+  var match = re.exec(url);
+
+  if (match != null) {
+    book_url = match[1] + "#toc";
+    downloadButtonCallback(book_url);
+  }
+  else {
+    renderStatus("Unknown url, not able to get book chapter url");
+  }
+}
+
 document.addEventListener('DOMContentLoaded', function() {
   var button = document.getElementById('download-button');
+  var loadAllInQueueButton = document.getElementById('load-all-in-queue-button');
   button.disabled = true;
+  loadAllInQueueButton.disabled = false;
 
   getCurrentTab(function(tab) {
-    var book_name = getBookNameFromUrl(tab.url);
-    if (book_name == null) {
-        setBookName("Unable to get book name, please change another url and try again")
-        return;
-    }
-    else {
-        setBookName(book_name);
-    }
+    var book_name;
 
-    // Inject content script and button callback
-    chrome.tabs.executeScript(null, {file: "content.js"});
+    chrome.tabs.onUpdated.addListener(function(tabId , info) {
+      if(tabId == tab.id && info.status == "complete"){
+        location.reload();
+        window.close();
+      }
+    });
 
-    var button = document.getElementById('download-button');
-    button.disabled = false;
-    button.addEventListener('click', function() {
-      tab_url = tab.url;
-      var re = new RegExp('(\.+://www.safaribooksonline.com/library/view/\.+/\\d+/)\.*');
-      var match = re.exec(tab_url)
-      if (match != null) {
-        book_url = match[1] + "#toc"
-        downloadButtonCallback(book_url);
+    loadAllInQueueButton.addEventListener('click', function() {
+      
+        chrome.tabs.executeScript( null, {code: "location.href='https://www.safaribooksonline.com/s/?limit=100'; "}, 
+          function() {
+            loadAllInQueueButton.textContent = "Loading...";
+            loadAllInQueueButton.disabled = true;
+          });
+      
+
+    });
+
+    if (isInQuenePage(tab.url))
+    {
+      loadAllInQueueButton.disabled = isInQuenePage(tab.url) == 2 ? true : false;
```

Comment:
If you allow user to download the book even when it's in a sub-section, e.g. https://www.safaribooksonline.com/s/?q=*&filter=Game%20Development&limit=30